File: ios/Classes/InAppStoryManagerAdaptor.swift
import Foundation
import Flutter
@_spi(IAS_API) import InAppStorySDK

class InAppStoryManagerAdaptor: InAppStoryManagerHostApi {
    init(binaryMessenger: FlutterBinaryMessenger) {
        InAppStoryManagerHostApiSetup.setUp(binaryMessenger: binaryMessenger, api: self)
    }
    
    func setPlaceholders(newPlaceholders: [String : String]) throws {
        InAppStory.shared.placeholders = newPlaceholders
    }
    
    func setTags(tags: [String]) throws {
        InAppStory.shared.settings?.tags = tags
        InAppStory.shared.setTags(tags)
    }
    
    func changeUser(userId: String, completion: @escaping (Result<Void, Error>) -> Void) {
        InAppStory.shared.settings = Settings(userID: userId)
        
        completion(.success(()))
    }
}

File: android/src/main/kotlin/com/example/inappstory_plugin/adaptors/InAppStoryManagerAdaptor.kt
package com.example.inappstory_plugin.adaptors

import InAppStoryManagerHostApi
import com.inappstory.sdk.InAppStoryManager
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding

class InAppStoryManagerAdaptor(
    flutterPluginBinding: FlutterPluginBinding,
    private val inAppStoryManager: InAppStoryManager,
) : InAppStoryManagerHostApi {

    init {
        InAppStoryManagerHostApi.setUp(flutterPluginBinding.binaryMessenger, this)
    }

    override fun setPlaceholders(newPlaceholders: Map<String, String>) {
        inAppStoryManager.placeholders = newPlaceholders
    }

    override fun setTags(tags: List<String>) {
        val arrayList = ArrayList<String>()
        arrayList.addAll(tags)
        inAppStoryManager.tags = arrayList
    }

    override fun changeUser(userId: String, callback: (Result<Unit>) -> Unit) {
        inAppStoryManager.userId = userId

        callback(Result.success(Unit))
    }

    override fun closeReaders() {
        InAppStoryManager.closeStoryReader(true, null)
    }
}